### PR TITLE
[v3.6.0] Root add recycle light func

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: #

### Changelog

* 

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by jk20012001 — APPROVED** [review]

File: cocos/core/root.ts [hunk 1]
```diff
@@ -649,11 +650,38 @@ export class Root {
...
+                break;
+            }
+        }
+        l.destroy();
+    }
+
+    /**
+     * @en recycle the given light to light object pool
+     * @zh 回收指定的光源到对象池
+     * @param l @en The light to be destroyed @zh 要回收的光源
```
Comment (pandamicro) [inline cocos/core/root.ts]: The light to be recycled

File: cocos/core/root.jsb.ts [hunk 2]
```diff
@@ -124,23 +124,22 @@ rootProto.createLight = function (LightCtor) {
 };
 
 rootProto.destroyLight = function (l) {
```
Comment (pandamicro) [inline cocos/core/root.jsb.ts:126]: Is recycleLight absent in native platform ?

File: cocos/core/root.ts [hunk 3]
```diff
@@ -649,11 +650,38 @@ export class Root {
...
+     */
+    public recycleLight (l: Light) {
```
Comment (pandamicro) [inline cocos/core/root.ts:676]: Please find `destroyLight` usage in Components and replace them with `recycleLight`

File: cocos/core/root.jsb.ts (hunk 2 above)
Comment (troublemaker52025) [reply cocos/core/root.jsb.ts:126]: Native platform uses this interface, so it needs to be modified synchronously

File: cocos/core/root.ts (hunk 3 above)
Comment (troublemaker52025) [reply cocos/core/root.ts:676]: sure

File: cocos/core/root.jsb.ts (hunk 2 above)
Comment (dumganhar) [reply cocos/core/root.jsb.ts:126]: Also needs recycleLight implementation in `root.jsb.ts`

**Review by pandamicro — APPROVED** [review]

**Comment by troublemaker52025** [comment]
@cocos-robot run test cases

**Review by dumganhar — APPROVED** [review]

**Pull request merged**